client-js/test/spec/tsTestSpec.ts

```diff
@@ -0,0 +1,42 @@
+import {Agent} from '../test/helloWorld';
+import {Observable} from '../app/scripts/observable';
+
+
+describe("Hello world", function() {
+  var agent;
+
+  beforeEach(function() {
+    agent = new Agent();
+  });
+
+  it("says hello", function() {
+    expect(agent.helloWorld()).toEqual("Hello world!");
+  });
+});
+
+/*
+* Test to create new observable and add callback
+* This is mostly to see if the plumbing for the
+* tests interfaces with the scripts in /app
+*/
+
+describe("observable", function() {
+  var observable;
+
+  beforeEach(function() {
+    observable = new Observable<any>();
+
+  });
+
+  function testFunc(message: string) {
+    console.log("testFunc called");
+    return message;
+  }
+
+
+  it("Add Callback", function() {
+    console.log("In callback test");
+    expect(observable.add(testFunc("something"))).toEqual("something");
```

Comment:
This should not compile. The code is passing a string to `observable.add`, but it wants a callback of type `any => void`, that is, a callback function. If the compiler is accepting this then there might be a problem with the way the compiler is invoked that it is not seeing the type definitions.